### ISSUE TITLE
Keyboard not in keyboard list (show keyboard button=on) on lockscreen in Android 10

### ISSUE THREAD

**Issue by mr-gittes** [AnySoftKeyboard/AnySoftKeyboard#2445]
### Steps to reproduce
1. activate anysoftkeyboard in android settings along with 2 other keyboards (stockrom + "other")
2. activate "show keyboard button" in settings
3. set lock screen password
4. lock screen => you see automatically stockrom keyboard (samsung etc.) when trying to write password
5. click on keyboard icon next to password input

### Expected behaviour
anysoftkeyboard and other keyboard should show up in list

### Actual behaviour
only other keyboard shows up!!!

(moreover: when I uninstall the stockrom keyboard, the other keyboard is the "main keyboard" and is automatically on lockscreen, not the anysoftkeyboard. when uninstalling the other keyboard, so there is only anysoftkeyboard left, the lockscreen shows no keyboard at all and no keyboard icon!! - you have to install stockrom or other keyboard again via adb to have  a keyboard at all... it looks like anysoftkeyboard  has no special authority to set itself on top of the list...the other keyboard and the stockrom keyboard can do it .... b.t.w. all permissions are set for anysoftkeyboard...

**Android OS version:**
10
**Device manufacturer and model:**
samsung, sm-n976b
**List of installed add-ons (like languages, or themes):**
-

**Comment by menny** [comment]
I can't reproduce this using Android 11. 
it might be an issue with Android 10?
Can you see if there are addition settings that need to be set?

I'll try Android 10 emulator later today

**Comment by menny** [comment]
I can't reproduce this at all.
Can you provide more details?

**Comment by Jonta** [comment]
@mr-gittes Poke. =)

**Comment by menny** [comment]
I will, yes

**Comment by menny** [comment]
Can anyone here reproduce this?

**Comment by menny** [comment]
@ItsIgnacioPortal and how do you recover from that?

**Comment by ItsIgnacioPortal** [comment]
> @ItsIgnacioPortal and how do you recover from that?

There are only a few ways of unlocking the device after that:

- Start an ADB shell from a machine that has a registered trusted ADB RSA fingerprint, and re-enable the google keyboard

- Remove the passcode entirely by flashing a file from a custom recovery (such as TWRP)

- Use a USB-OTG keyboard to enter the passcode

TL;DR: If you have no trusted ADB RSA fingerprints, no custom recovery, and no USB OTG support, you're screwed.

Luckily, I was able to use a keyboard over USB OTG, as I already had the special cable that's needed. I had to connect a USB hub to the USB OTG cable to make it work.

**Comment by menny** [comment]
Oh crap. So sorry.

**Comment by menny** [comment]
I have a proposed fix (I'm not sure it will fix the problem, since it seems that different devices behave, well, differently).
Is anyone willing to test it out? Please do so only if you have a way to type in the passcode in case this fix is not valid.

**Comment by ItsIgnacioPortal** [comment]
I would be glad to test it. Just compile the apk for me

**Comment by menny** [comment]
There is another fix that needs to be done. I'll post the APK once it is ready. Thanks

**Comment by menny** [comment]
[ime-app-7388.zip](https://github.com/AnySoftKeyboard/AnySoftKeyboard/files/8486057/ime-app-7388.zip) <- Test APK with direct-boot and lock-screen support.

**Comment by ItsIgnacioPortal** [comment]
Alright, so, a couple remarks:

- You can test this safely by just not disabling the Google Keyboard. Now, there's a little keyboard icon on the lockscreen that allows me to select _AnySoftKeyboard Canary_ as the keyboard:

![Screenshot_20220414-212753](https://user-images.githubusercontent.com/46269721/163497623-659c5e77-181a-44ca-a3f5-78a261f69809.png)

![2022-04-14 21_31_19-Window](https://user-images.githubusercontent.com/46269721/163497717-444d7aa4-c767-47cf-9ff7-515696fcac29.png)

- On the lockscreen, after selecting ASK Canary in the keyboards list, the keyboard doesn't actually appear, so I can't type-in the password.
- For whathever reason, using a USB OTG keyboard keyboard doesn't work while ASK Canary is selected in the lockscreen. I had to reboot my device, and then the USB OTG keyboard started working again; With GBoard. And yes, this issue is unique to the lockscreen. I have used a USB OTG keyboard before with ASK, and it worked well.
- The USB OTG keyboard works fine on the lockscreen with ASK _after_ the initial unlock.

**Comment by billie80** [comment]
I also had this with the version above. @menny already committed a few changes fixing it. He shared an updated version privatly so I am not sure I can add it here.

**Comment by menny** [comment]
@billie80 please share it here.

**Comment by billie80** [comment]
Okay, here we go!
[ime-app-7390.zip](https://github.com/AnySoftKeyboard/AnySoftKeyboard/files/8495855/ime-app-7390.zip)

**Comment by ItsIgnacioPortal** [comment]
It works!

Good job guys

**Comment by pjmkrpg** [comment]
Thanks @billie80, your zip contains the only version of anysoftkeyboard that allows me to unlock my phone on reboot (other versions won't run on startup...). However it seems not to be compatible with language pack from f-droid: the language works until reboot then disappears from app's settings, I must run f-droid to uninstall and reinstall it. How can I solve the problem?

**Comment by miphilip** [comment]
I wonder why the zip from @billie80 isnt officially released as its the only one so far solving this issue but yet hard to find

**Comment by billie80** [comment]
May I ask why it is difficult? Just curious!
